pkg/hive/cell/lifecycle.go

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cell
+
+import "context"
+
+// Lifecycle is a interface for using hive.Lifecycle without importing hive or depending on
+// any other hive types.
+// The default lifecycle implementation is in hive/lifecycle.go implements this interface
+// using the existing hook functions, and it is provided under this interface to dissociate
+// the two types.
+//
+// This is used by cell.Module to register Module level hooks, without creating a circular
+// dependency between cell and hive, as well as not requiring other implementations of hive.Lifecycle
+// to provide this functionality.
+//
+// The hook functions should be translated into the appropriate hook interface, and the hook context
+// is cast into a regular context.Context.
+//
+// This should only be used by cell internals, other packages should use hive.Lifecycle instead.
+type Lifecycle interface {
```

Comment:
@joamaki This doesn't feel great - but I couldn't find a better way to do this without refactoring a bunch of things. Not sure if you have any better ideas here 🤔

Reply:
Let's try moving types from `pkg/hive/lifecycle.go` to `pkg/hive/cell/lifecycle.go` (without `DefaultLifecycle` though), and then adding type aliases to `pkg/hive/lifecycle.go` to keep everything working. This would also be nice in that we'll then only need to import `cell` package when creating new cells and not both `cell` and `hive`.